source/common/router/config_impl.cc

```diff
@@ -664,15 +672,29 @@ std::string RouteEntryImplBase::newPath(const Http::RequestHeaderMap& headers) c
     final_host = processRequestHost(headers, final_scheme, final_port);
   }
 
+  std::string final_path_value;
+  bool path_redirect_has_query = false;
```

Comment:
Last pass. I should have realized this earlier, but the value we compute for this variable never changes for a given instance of RouteEntryImplBase. I think this should become a const member variable of RouteEntryImplBase. It'll save us a string search on every request that hits a path redirect.